include/hal/library/responder_secretlib.h

```diff
@@ -270,4 +270,29 @@ extern bool libspdm_gen_csr(uint32_t base_hash_algo, uint32_t base_asym_algo, bo
                             size_t *csr_len, uint8_t *csr_pointer);
 #endif /* LIBSPDM_ENABLE_CAPABILITY_GET_CSR_CAP */
 
+#if LIBSPDM_ENABLE_CAPABILITY_SET_CERT_CAP
+/**
+ * return if current code is running in a trusted environment.
+ *
+ * @retval  true   It is in a trusted environment.
+ * @retval  false  It is not in a trusted environment.
+ **/
+extern bool libspdm_is_in_trusted_environment();
+
+/**
+ * Stores a certificate chain in non-volatile memory.
+ *
+ *
+ * @param[in]  slot_id          The number of slot for the certificate chain.
+ * @param[in]  cert_chain       The pointer for the certificate chain to set.
+ * @param[in]  cert_chain_size  The size of the certificate chain to set.
+ *
+ * @retval true   The certificate chain was successfully written to non-volatile memory.
+ * @retval false  Unable to write certificate chain to non-volatile memory.
+ **/
+extern bool libspdm_write_certificate_to_nvm(uint8_t slot_id, const void * cert_chain,
```

Comment:
Why move this to secretlib? Certificates aren't secret.

Reply:
All certificate are managed by secretlib.

platformlib deal with watchdog / timer, etc.

Reply:
Writing flash requires integrity guarantee. Generic platformlib is not a good place.

At least it is handled by [spdm_device_secret_lib_sample](https://github.com/DMTF/libspdm/tree/main/os_stub/spdm_device_secret_lib_sample) today, because all certificate management is done by secretlib.

Reply:
@xiaoyuruan can you be the tie-breaker? Should the code that writes the certificate chain to FLASH live in the secure enclave or in the platform library with timers and watchdogs?

Reply:
According to the current definition of spdm_device_secret_lib

[spdm_device_secret_lib](https://github.com/DMTF/libspdm/blob/main/include/library/spdm_device_secret_lib.h)

This library handles the private key signing, PSK HMAC operation, and measurement collection.

I agree that "certificate write to FLASH" does not fall under this definition.

Reply:
[platform_lib](https://github.com/DMTF/libspdm/blob/main/include/hal/library/platform_lib.h) provides sleep function and watchdog function, 

I don't think it belongs to platform_lib.

We can have below options:
1) redefine secret lib
2) redefine platform lib
3) define a new lib

Reply:
We can have a separate issue to sort out the libraries and headers for the 3.0 release.

Reply:
agree. I filed https://github.com/DMTF/libspdm/issues/2124